app/controllers/assignments_controller.rb

```diff
@@ -15,6 +15,7 @@ def show
 
   def new
     @assignment = Assignment.new
+    @assignment.requirement = @requirement
```

Comment:
Isn't this covered by line 5, the before hook ?

Reply:
Ah, I didn't parse this correctly the first time. The before hook **is** defining the requirement, and here you're making it the parent to the new assignment.